contracts/MainStaking.sol

```diff
@@ -3,11 +3,27 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+// import "@openzeppelinUpgradeable/contracts/proxy/utils/Initializable.sol";
+
 import "./../contracts/xQI.sol";
-import "./../contracts/QI.sol";
 
 /// @title MainStaking
+/// @notice Mainstaking is the contract that interacts with ALL xQi contract
 
 contract MainStaking {
+    using SafeERC20 for IERC20;
+    xQI public xqi;
+    address public immutable stakingToken;
+
+    constructor (address _stakingToken) {
+        stakingToken = _stakingToken;
+    }
+
+    event ClaimApproved(address user, uint256 amount, bool isApproved);
 
+    /// @notice claimApprove checks xQi balance of the sender then call claim in BaseRewardPool
+    function claimApprove(uint256 _userBalance) external {
+        require(_userBalance > 0, "xQi balance must be positive");
+        emit ClaimApproved(msg.sender, _userBalance, true);
+    }
```

Comment:
Remember  : the user cannot be trusted for these inputs, the caller must be a contract that we own

Reply:
I added an onlyOwner modifier